File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    let result = 0;
    while (num !== 1 && result !== 500) {
        num = num % 2 === 0 ? num / 2 : num * 3 + 1;
        result++;
    }
    return result >= 500 ? -1 : result;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int result = 0;
        while (num != 1) {
            if (num % 2 == 0) num /= 2;
            else if (num % 2 == 1) num = num * 3 + 1;
            result++;
            
            if (result > 500 || num < 0) {
                result = -1;
                break;
            }
        }
        return result;
    }
}